### ISSUE TITLE
Neovim support

### ISSUE THREAD

**Issue by reaysawa** [skywind3000/vim-tvision#1]
Hello! Would you accept contributions for also supporting Neovim?

The way I imagine it is by creating a subdirectory `lib` with the files
```
- lib/vim.vim
- lib/neovim.vim
```

Then sourcing **lib/neovim.vim** at runtime `if has('nvim')`.

I've asked in the Neovim Gitter channel and there are currently no plans for mimicking the Vim `popup_*` API.

---

For Neovim support to be enabled, the code would also have to change to something like this:

https://github.com/skywind3000/vim-tvision/blob/master/autoload/tvision.vim#L277

```patch
- let winid = popup_create([], opts)
+ let winid = tvision#lib#popup_create([], opts)
- call popup_hide(inst.win_popup)
+ call tvision#lib#popup_hide(inst.win_popup)
- call win_execute(inst.win_popup, 'setlocal nonumber nowrap')
+ call tvision#lib#win_execute(inst.win_popup, 'setlocal nonumber nowrap')
...
```

Basically change the direct Vim API calls to functions with different float windows code for both Vim and Neovim.

---

I'm asking because this would change the code organization drastically. It's understandable if you don't want that. Let me know!

**Comment by reaysawa** [comment]
We already have some alternatives on Neovim's side so I'm closing this one. Sorry to bother.
